src/backend/distributed/shared_library_init.c

```diff
@@ -997,6 +1026,74 @@ NormalizeWorkerListPath(void)
 }
 
 
+static bool
+NodeConninfoGucCheckHook(char **newval, void **extra, GucSource source)
+{
+	/* this array _must_ be kept in an order usable by bsearch */
+	const char *whitelist[] = {
+		"application_name",
+		"connect_timeout",
+			#if defined(ENABLE_GSS) && defined(ENABLE_SSPI)
+		"gsslib",
+			#endif
+		"keepalives",
+		"keepalives_count",
+		"keepalives_idle",
+		"keepalives_interval",
+			#if defined(ENABLE_GSS) || defined(ENABLE_SSPI)
+		"krbsrvname",
+			#endif
+		"sslcompression",
+		"sslcrl",
+		"sslmode",
+		"sslrootcert"
+	};
+	char *errorMsg = NULL;
+	bool conninfoValid = CheckConninfo(*newval, whitelist, lengthof(whitelist),
+									   &errorMsg);
+
+	if (!conninfoValid)
+	{
+		GUC_check_errdetail("%s", errorMsg);
+	}
+
+	return conninfoValid;
+}
+
+
+static void
+NodeConninfoGucAssignHook(const char *newval, void *extra)
+{
+	PQconninfoOption *optionArray = NULL;
+	PQconninfoOption *option = NULL;
+
+	if (newval == NULL)
+	{
+		newval = "";
+	}
+
+	optionArray = PQconninfoParse(newval, NULL);
+	if (optionArray == NULL)
+	{
+		ereport(ERROR, (errmsg("Unexpected")));
```

Comment:
The user we'll probably never see this, but for the developer maybe add a comment / message on why it's unexpected. e.g. `"Unexpected because NodeConninfoGucCheckHook prevents invalid values"

Reply:
Yeah I almost considered making it an `Assert`; what are your thoughts on that?

If I keep it as an `ereport`, what do you think about making it `FATAL`? It seems like a pretty bad state to be in. I'll update the error language regardless.

Reply:
I have a slight dislike of asserts because they're not so easy to debug and create slight behavioural changes, FATAL makes sense.

Reply:
Changed.